packages/headless/src/controllers/did-you-mean/headless-did-you-mean.ts

```diff
@@ -68,13 +71,27 @@ export function buildDidYouMean(
 
   dispatch(enableDidYouMean());
 
+  const rebuildOriginalQuery = (state: DidYouMeanSection) => {
+    return state.didYouMean.queryCorrection.correctedQuery
+      .split(' ')
+      .reduce((current, possiblyCorrected, idx) => {
+        const correctionFound = state.didYouMean.queryCorrection.wordCorrections.find(
+          (correction) => correction.correctedWord === possiblyCorrected
+        );
+        return `${current}${idx === 0 ? '' : ' '}${
+          correctionFound ? correctionFound.originalWord : possiblyCorrected
+        }`;
+      }, '');
+  };
```

Comment:
Instead of reconstructing the original query, could we instead hold the value in the DYM portion of the state? I see there is already a reducer for the `executeSearch.fulfilled` case. The payload of that action has a `queryExecuted` property, but this action may not work because the auto correction mechanism performs a second search, which would overwrite the value. 

We could create a new action `didYouMeanCorrectionReceived`, responsible for writing the original query to state.

My concern with the current approach is whether there is some edge case, a language, where the above code would not work. Not sure if it is possible, only a feeling, but if it's possible to avoid, then why not?